app/main/appium-method-handler.js

```diff
@@ -213,4 +215,40 @@ export default class AppiumMethodHandler {
     }
   }
 
+  static appiumHandlers = {}
+
+  static createSession (driver, sender, winId) {
+    const {appiumHandlers} = AppiumMethodHandler;
+    log.info(`Creating method handler for session with window id: ${winId}`);
+    const handler = AppiumMethodHandler.appiumHandlers[winId] = new AppiumMethodHandler(driver, sender);
+    log.info(`The following session window ID's have active sessions: '${JSON.stringify(_.keys(appiumHandlers))}'`);
+    return handler;
+  }
+
+  static killSession (winId, killedByUser) {
+    const {appiumHandlers} = AppiumMethodHandler;
```

Comment:
why recreate all the `getSessionHandler` logic in here?

Reply:
There was a race condition where the appiumHandlers sessions were being prematurely deleted. So to debug, I moved all the logic into that one class and then added logging to it so I can see when objects in the key were being get/set/deleted.

Reply:
nvm, i understand what you're talking about now